docs/en/week06/06-3.md

```diff
@@ -221,38 +211,77 @@ How do we turn off the output? Let’s say we have a purple internal representat
 <b>Figure 18:</b>LSTM Architecture - Output Off
 </center>
 
-
 Similarly, we can control the memory. For example, we can reset it by having $f[t]$ and i[t] to be zeros. After multiplication and summation, we have a zero inside the memory. Otherwise, we can keep the memory, by still zeroing out the internal representation $th$ but keep a one in $f[t]$. Hence, the sum gets $c[t-1]$ and keeps sending it out. Finally, we can write such that we can get a one in the input gate, the multiplication gets purple, then set a zero in the don’t forget gate so it actually forget.
+
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/memory_cell_vis.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>Figure 19:</b>Visualization of the Memory Cell
+</center>
+
 <center>
 <img src="{{site.baseurl}}/images/week06/06-3/lstm_4.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
-<b>Figure 19:</b>LSTM Architecture - Reset Memory
+<b>Figure 20:</b>LSTM Architecture - Reset Memory
+</center>
+
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/lstm_keep_memory.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>Figure 21:</b>LSTM Architecture - Keep Memory
 </center>
 
+<center>
+<img src="{{site.baseurl}}/images/week06/06-3/lstm_write_memory.png" style="zoom: 40%; background-color:#DCDCDC;"/><br>
+<b>Figure 22:</b>LSTM Architecture - Write Memory
+</center>
 
 ## Notebook Examples
 
 ### Sequence Classification
 
-The goal is to classify sequences. Elements and targets are represented locally (input vectors with only one non-zero bit). The sequence starts with an B, ends with a E (the “trigger symbol”), and otherwise consists of randomly chosen symbols from the set {a, b, c, d} except for two elements at positions t1 and t2 that are either X or Y. For the DifficultyLevel.HARD case, the sequence length is randomly chosen between 100 and 110, t1 is randomly chosen between 10 and 20, and t2 is randomly chosen between 50 and 60. There are 4 sequence classes Q, R, S, and U, which depend on the temporal order of X and Y. The rules are: X, X -> Q; X, Y -> R; Y, X -> S; Y, Y -> U.
+The goal is to classify sequences. Elements and targets are represented locally (input vectors with only one non-zero bit). The sequence starts with an $B$, ends with a $E$ (the “trigger symbol”), and otherwise consists of randomly chosen symbols from the set $\{a, b, c, d\}$ except for two elements at positions $t1$ and $t2$ that are either $X$ or $Y$. For the DifficultyLevel.HARD case, the sequence length is randomly chosen between 100 and 110, $t1$ is randomly chosen between 10 and 20, and $t2$ is randomly chosen between 50 and 60. There are 4 sequence classes $Q, R, S$, and $U$, which depend on the temporal order of $X$ and $Y$. The rules are: `X, X -> Q`; `X, Y -> R`; `Y, X -> S`; `Y, Y -> U`.
```

Comment:
```suggestion
The goal is to classify sequences. Elements and targets are represented locally (input vectors with only one non-zero bit). The sequence **b**egins with an `B`, **e**nds with a `E` (the “trigger symbol”), and otherwise consists of randomly chosen symbols from the set `{a, b, c, d}` except for two elements at positions $t_1$ and $t_2$ that are either `X` or `Y`. For the `DifficultyLevel.HARD` case, the sequence length is randomly chosen between 100 and 110, $t_1$ is randomly chosen between 10 and 20, and $t_2$ is randomly chosen between 50 and 60. There are 4 sequence classes `Q`, `R`, `S`, and `U`, which depend on the temporal order of `X` and `Y`. The rules are: `X, X -> Q`; `X, Y -> R`; `Y, X -> S`; `Y, Y -> U`.
```